src/infrasys/time_series_manager.py

```diff
@@ -280,20 +320,66 @@ def deserialize(
         """Deserialize the class. Must also call add_reference_counts after deserializing
         components.
         """
-        time_series_dir = Path(parent_dir) / data["directory"]
-
-        if _process_time_series_kwarg("time_series_read_only", **kwargs):
-            storage = ArrowTimeSeriesStorage.create_with_permanent_directory(time_series_dir)
-        else:
-            storage = ArrowTimeSeriesStorage.create_with_temp_directory()
-            storage.serialize(src=time_series_dir, dst=storage.get_time_series_directory())
-
-        cls_instance = cls(con, storage=storage, initialize=False, **kwargs)
+        if (
+            _process_time_series_kwarg("time_series_storage_type", **kwargs)
+            == TimeSeriesStorageType.MEMORY
+        ):
+            msg = "De-serialization does not support in-memory time series storage."
+            raise ISOperationNotAllowed(msg)
 
-        if _process_time_series_kwarg("time_series_in_memory", **kwargs):
-            cls_instance.convert_storage(**kwargs)
+        dst_time_series_directory = _process_time_series_kwarg("time_series_directory", **kwargs)
+        if dst_time_series_directory is not None and not Path(dst_time_series_directory).exists():
+            msg = f"time_series_directory={dst_time_series_directory} does not exist"
+            raise FileNotFoundError(msg)
+        read_only = _process_time_series_kwarg("time_series_read_only", **kwargs)
+        time_series_dir = Path(parent_dir) / data["directory"]
+        storage: TimeSeriesStorageBase
+
+        match data["time_series_storage_type"]:
```

Comment:
Just realized that this needs a change to handle existing files. I'll push one more minor commit soon.